### ISSUE TITLE
Confusion around the topic names and settings involving topics/channels

### ISSUE THREAD

**Issue by safandi-personal** [iainporter/sms-service#7]
In your readme file you have command to install the connector with one of the options as:
"table.whitelist": "public.outboxevent",
so when i check the list of topics i see topic with <serverid>.<schema>.<tablename> so in it will be:
smsdb1.public.outboxevent is the topic name i see using kafka-topics.sh

however in the application.properties file the settings related to consumer are:
mp.messaging.incoming.sms.connector=smallrye-kafka
mp.messaging.incoming.sms.topic=**outbox.event.sms_message**

Can you please explain why the topic name is different in the application from what i see on the kafka side?

Also another issue i faced was the config for installing connector. the following dont work for me:
"transforms" : "outbox","transforms.outbox.type" : "io.debezium.transforms.outbox.EventRouter", "transforms.OutboxEventRouter.event.key": "aggregate_id", 
"transforms.outbox.table.fields.additional.placement": "type:header:eventType"

When i install the connector without these parameters it works fine and i can see the payload in the topic smsdb1.public.outboxevent with schema. 
When i set 
key.converter.schemas.enable=false
value.converter.schemas.enable=false
in the properties file then the payload in topic is without Schema

So Basically i am confused about why the connector topic settings differ from the application.properties settings in terms of names?
Isnt it suppose to consume events from the topic which is created on kafka? If yes why are the topic name values different?
Would really love to understand how all of this works. Thanks.

**Comment by iainporter** [comment]
I found it a bit confusing myself and got it working with trial and error

My main reference was this https://debezium.io/documentation/reference/1.5/configuration/outbox-event-router.html

and

https://debezium.io/documentation/reference/connectors/postgresql.html

**Comment by safandi-personal** [comment]
Is there anything specific u want me to look at?. I have read those links many times and i still cant completely understand how this works and what am i doing wrong or missing. 
Infact I end up with more questions. Where do you specify these settings:
transforms=outbox
transforms.outbox.type=io.debezium.transforms.outbox.EventRouter
transforms.outbox.table.fields.additional.placement=type:header:eventType

Also why dont they work for me when i add them as part of my connector postgresql config?

**Comment by safandi-personal** [comment]
> 
> 
> I found it a bit confusing myself and got it working with trial and error
> 
> My main reference was this https://debezium.io/documentation/reference/1.5/configuration/outbox-event-router.html
> 
> and
> 
> https://debezium.io/documentation/reference/connectors/postgresql.html

Ok I was able to resolve my issue finally. So all my code, kafka, kafka connect, postgres db setup via docker-compose yaml file is custom. 
I am using the "debezium-examples/outbox" as reference. My versions are also newer than the ones mentioned in "outbox" git example.
I had to install "Kafka connect UI" to figure out the problem. Based on debezium documentation it tells you how to install plug-ins for your desired database. However it doesn't mention anything about other required libs/jar files needed to run the "outbox" transformation configuration in the connector.  Through the "kafka connect ui" after applying the config:

curl -s -X POST -H "Accept:application/json" -H "Content-Type:application/json" localhost:8083/connectors/ -d '{
"name": "user-connector",
"config": {
      "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
      "tasks.max": "1",
      "database.hostname": "user-db",
      "database.port": "5432",
      "database.user": "postgresuser",
      "database.password": "postgrespw",
      "database.dbname" : "userdb",
      "database.server.name": "dbserver1",
      "table.whitelist" : "public.outboxevent",
      "tombstones.on.delete": "false",
      "transforms" : "outbox",
      "transforms.outbox.type" : "io.debezium.transforms.outbox.EventRouter",
      "transforms.outbox.route.topic.replacement" : "contactinfo.events",
      "transforms.outbox.table.fields.additional.placement" : "id:header"}
}'

I saw Error in the "trace" section. It was complaining about **"io.opentracing"** classes not found. So I had to wget those jar files and install them in the plugin directory. After that I restarted my kafka connect, my application,  and it worked exactly like "outbox" Git example. 
I wish the documentation around this is more clear as not all images of kakfa connect have all the required jar files in order for SMT EvenRouter to work. Maybe some images have all the necessary jar files inside the plugin directory but that wasn't the case for my installation.
